compiler/rustc_resolve/src/lib.rs

```diff
@@ -660,6 +660,8 @@ impl<'a> fmt::Debug for ModuleData<'a> {
 struct NameBinding<'a> {
     kind: NameBindingKind<'a>,
     ambiguity: Option<(&'a NameBinding<'a>, AmbiguityKind)>,
```

Comment:
```suggestion
    ambiguity: Option<(&'a NameBinding<'a>, AmbiguityKind, bool)>,
```
The warning flag only makes sense if there is ambiguity.